invenio_users_resources/permissions.py

```diff
@@ -1,13 +1,75 @@
 # -*- coding: utf-8 -*-
 #
 # Copyright (C) 2023 CERN.
+# Copyright (C) 2025 Ubiquity Press.
 #
 # Invenio-Users-Resources is free software; you can redistribute it and/or modify
 # it under the terms of the MIT License; see LICENSE file for more details.
 """Users resources generic needs and permissions."""
 
-from invenio_access import action_factory
+from invenio_access import ActionRoles, Permission, action_factory, superuser_access
+from invenio_records_permissions.generators import AdminAction
+from invenio_search.engine import dsl
 
 USER_MANAGEMENT_ACTION_NAME = "administration-moderation"
 
 user_management_action = action_factory(USER_MANAGEMENT_ACTION_NAME)
+
+
+class SuperUserMixin:
+    """Mixin for superuser permissions."""
+
+    def _is_group_superadmin(self, group_id, roles):
+        groups = {role.role.id for role in roles}
+        if group_id in groups:
+            return True
+        return False
+
+    def _get_superadmin_roles(self):
+        """Get all roles with superuser access action role."""
+        return ActionRoles.query_by_action(superuser_access).all()
+
+    def _is_user_superadmin(self, identity, roles=None):
+        """Check if the user has the superuser role."""
+        roles = roles if roles else self._get_superadmin_roles()
+        users = {user.id for role in roles for user in role.role.users}
+        if identity.id in users:
+            return True
+        return False
+
+    def _should_action_proceed(self, identity, group):
```

Comment:
Why can't we include this (or some form of it) in `def needs` ... `def _condition` ... etc.?